src/table/index.ts

```diff
@@ -0,0 +1,378 @@
+const cellActionsClassName = "kuc-table__table__body__row__action";
+const btnAddRowClassName = "kuc-table__table__body__row__action-add";
+const btnRemoveRowClassName = "kuc-table__table__body__row__action-remove";
+
+const dAdd =
+  "M8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16ZM12.0355 8.49997V7.49997H8.50008V3.96454H7.50008V7.49997H3.96443V8.49997H7.50008V12.0356H8.50008V8.49997H12.0355Z";
+const fillAdd = "#3498db";
+const dRemove =
+  "M16 8C16 12.4183 12.4183 16 8 16C3.58172 16 0 12.4183 0 8C0 3.58172 3.58172 0 8 0C12.4183 0 16 3.58172 16 8ZM12.0355 7.49997V8.49997L3.96443 8.49997V7.49997H12.0355Z";
+const fillRemove = "#d4d7d7";
+
+let exportTable;
+(() => {
+  exportTable = window.customElements.get("kuc-table");
+  if (exportTable) {
+    return;
+  }
+  class KucTable extends KucBase {
+    @property({ type: String, reflect: true, attribute: "class" }) className =
+      "";
+    @property({ type: String, reflect: true, attribute: "id" }) id = "";
+    @property({ type: String }) label = "";
+    @property({ type: Array }) columns: Column[] = [];
+    @property({ type: Array }) data: object[] = [];
+    @property({ type: Boolean }) actionButton = true;
+    @property({
+      type: Boolean,
+      attribute: "hidden",
+      reflect: true,
+      converter: visiblePropConverter,
+    })
+    visible = true;
+
+    @query(".kuc-table__table")
+    private _table!: HTMLTableElement;
+
+    @query(".kuc-table__table__body")
+    private _tBody!: HTMLTableSectionElement;
+
+    constructor(props?: TableProps) {
+      super();
+      if (!props) return;
+
+      const validProps = validateProps(props);
+      Object.assign(this, validProps);
+    }
+
+    protected shouldUpdate(_changedProperties: PropertyValues): boolean {
+      if (_changedProperties.has("columns")) {
+        const errorMessage = this._getErrorMessageWhenValidateColumns();
+        if (errorMessage) {
+          throwErrorAfterUpdateComplete(this, errorMessage);
+          return false;
+        }
+      }
+      if (_changedProperties.has("data") && !validateArrayProperty(this.data)) {
+        const errorMessage = ERROR_MESSAGE.DATA_TABLE.IS_NOT_ARRAY;
+        throwErrorAfterUpdateComplete(this, errorMessage);
+        return false;
+      }
+      return true;
+    }
+
+    protected willUpdate(_changedProperties: PropertyValues): void {
+      if (!this._tBody) return;
+      this._tBody.innerHTML = "";
+    }
+
+    private _getErrorMessageWhenValidateColumns() {
+      if (!validateArrayProperty(this.columns)) {
+        return ERROR_MESSAGE.COLUMNS.IS_NOT_ARRAY;
+      }
+      if (!validateFieldRequiredInColumnTable(this.columns)) {
+        return ERROR_MESSAGE.COLUMNS.FIELD_REQUIRED;
+      }
+      if (validateFieldUniqueInColumnTable(this.columns)) {
+        return ERROR_MESSAGE.COLUMNS.FIELD_UNIQUE;
+      }
+      return "";
+    }
+
+    render() {
+      return html`
```

Comment:
return !this.columns || this.columns.length < 1
        ? html`
            <table class="kuc-table__table">
              <caption
                class="kuc-table__table__label"
                ?hidden="${!this.label}"
              >
                ${this.label}
              </caption>
            </table>
          `
        : html` .....`